cms/html.py

```diff
@@ -25,76 +19,65 @@ def process(text):
-                    except (permalinks.PermalinkError, ObjectDoesNotExist):
-                        resolved_permalinks[value] = None
-                obj = resolved_permalinks[value]
-                if obj:
-                    # Add in the URL of the obj.
-                    attrs[attr_name] = '"%s"' % escape(obj.get_absolute_url())
-                    # Add in the title of the obj.
-                    attrs.setdefault("title", '"%s"' % escape(getattr(obj, "title", str(obj))))
-                return obj
-            return None
-
-        if tagname == "a":
-            # Process hyperlinks.
-            obj = get_obj("href")
-        elif tagname == "img":
-            # Process images.
-            obj = get_obj("src")
-
-            if obj:
-                if hasattr(obj, 'attribution') or hasattr(obj, 'copyright'):
-                    attrs["title"] = ''
-
-                    if hasattr(obj, 'copyright') and obj.copyright:
-                        attrs["title"] += '&copy; {}. '.format(
-                            obj.copyright,
-                        )
-
-                    if hasattr(obj, 'attribution') and obj.attribution:
-                        attrs["title"] += obj.attribution
-
-                    if attrs["title"]:
-                        attrs["title"] = '"{}"'.format(attrs["title"])
+                        thumbnail = get_thumbnail(getattr(obj, fieldname), '{}x{}'.format(width, height), quality=99, format="PNG")
+                    except IOError:
+                        pass
                     else:
-                        attrs["title"] = '"{}"'.format(obj.title)
+                        image['src'] = '%s' % escape(thumbnail.url)
+                        image['width'] = '%s' % thumbnail.width
+                        image['height'] = '%s' % thumbnail.height
 
-                try:
-                    width = int(attrs["width"][1:-1])
-                    height = int(attrs["height"][1:-1])
-                except (ValueError, KeyError, TypeError):
-                    pass
-                else:
-                    # Automagically detect a FileField.
-                    fieldname = None
-                    for field in obj._meta.fields:
-                        if isinstance(field, models.FileField):
-                            fieldname = field.name
-                    # Generate the thumbnail.
-                    if fieldname:
-                        try:
-                            thumbnail = get_thumbnail(getattr(obj, fieldname), '{}x{}'.format(width, height), quality=99, format="PNG")
-                        except IOError:
-                            pass
-                        else:
-                            attrs["src"] = '"%s"' % escape(thumbnail.url)
-                            attrs["width"] = '"%s"' % thumbnail.width
-                            attrs["height"] = '"%s"' % thumbnail.height
-        else:
-            assert False
+    return str(soup.decode(formatter=None))
```

Comment:
Should `six.text_type(soup)` be fine here? (Or `str(soup)` whenever we remove Python 2 support.)

Reply:
From the [docs](https://www.crummy.com/software/BeautifulSoup/bs4/doc/#output-formatters):

> If you pass in formatter=None, Beautiful Soup will not modify strings at all on output. This is the fastest option, but it may lead to Beautiful Soup generating invalid HTML/XML[.]

Reply:
it's formatter=None at the moment as otherwise soup turns the `&copy;` into `&amp;copy;`